src/script/components/MessagesList/MessageList.tsx

```diff
@@ -245,78 +255,167 @@ export const MessagesList: FC<MessagesListParams> = ({
+                  hasReadReceiptsTurnedOn={conversationRepository.expectReadReceipt(conversation)}
+                  isLastDeliveredMessage={isLastDeliveredMessage}
+                  isHighlighted={isHighlighted}
+                  scrollTo={scrollToElement}
+                  isSelfTemporaryGuest={selfUser.isTemporaryGuest()}
+                  messageRepository={messageRepository}
+                  onClickAvatar={showUserDetails}
+                  onClickCancelRequest={cancelConnectionRequest}
+                  onClickImage={showImageDetails}
+                  onClickInvitePeople={() => invitePeople(conversation)}
+                  onClickReactionDetails={message => showMessageReactions(message, true)}
+                  onClickMessage={onClickMessage}
+                  onClickParticipants={showParticipants}
+                  onClickDetails={message => showMessageDetails(message)}
+                  onClickResetSession={resetSession}
+                  onClickTimestamp={async function (messageId: string) {
+                    setHighlightedMessage(messageId);
+                    setTimeout(() => setHighlightedMessage(undefined), 5000);
+                    const messageIsLoaded = conversation.getMessage(messageId);
+
+                    if (!messageIsLoaded) {
+                      const messageEntity = await messageRepository.getMessageInConversationById(
+                        conversation,
+                        messageId,
+                      );
+                      conversation.removeMessages();
+                      conversationRepository.getMessagesWithOffset(conversation, messageEntity);
+                    }
+                  }}
+                  selfId={selfUser.qualifiedId}
+                  shouldShowInvitePeople={shouldShowInvitePeople}
+                  isFocused={isFocused}
+                  handleFocus={setFocusedId}
+                  handleArrowKeyDown={handleKeyDown}
+                  isMsgElementsFocusable={isMsgElementsFocusable}
+                  setMsgElementsFocusable={setMsgElementsFocusable}
+                />
+              );
+            });
+          })}
+        </div>
+      </FadingScrollbar>
+      <JumpToLastMessageButton onGoToLastMessage={jumpToLastMessage} conversation={conversation} />
+    </>
+  );
+};
+
+export interface JumpToLastMessageButtonProps extends HTMLProps<HTMLElement> {
+  onGoToLastMessage: () => void;
+  conversation: Conversation;
+}
+
+export const JumpToLastMessageButton: FC<JumpToLastMessageButtonProps> = ({
+  onGoToLastMessage,
+  conversation,
+}: JumpToLastMessageButtonProps) => {
+  // To be changed when design chooses a breakpoint, the conditional can be integrated to the ui-kit directly
+  const mdBreakpoint = useMatchMedia('max-width: 768px');
+
+  const [isLastMessageVisible, setIsLastMessageVisible] = useState(conversation.isLastMessageVisible());
+
+  useEffect(() => {
+    const subscription = conversation.isLastMessageVisible.subscribe(
```

Comment:
I can't use here `useKoSubscribableChildren` as I need to add debouncing to the change of the state.

Reply:
I tried it, but there is also one more thing - the initial value of `isLastMessageVisible` should be set by the last fired value. Just waiting for the new event and debouncing makes it behave really weird. I suggest to keep my existing implementation